app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -145,31 +159,60 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
+     * @param followedId user who is being sent the request
+     * @param followerId user who sent the request
+     * @return true if user already sent the request
+     */
+    private boolean searchInRequestsList(String followedId, String followerId) {
+        // retrieving document of followed user
+        DocumentReference userDocument = _db.collection("Users").document(followedId);
+        Task<DocumentSnapshot> querySnapshotTask = userDocument.get();
+
+        while (!querySnapshotTask.isComplete());
+        Map<String, Object> objectMap = querySnapshotTask.getResult().getData();
+        ArrayList<String> requestList = (ArrayList<String>) objectMap.get(FOLLOW_REQUEST_FIELD);
+        if (requestList != null && !requestList.isEmpty()) {
+            return requestList.contains(followerId);
+        }
+        return false;
+    }
+
+    /**
+     * Checks whether a user has followed another user already
+     * @param followedId user being sent follow request
+     * @param followerId user sending follow request
+     * @return true if user has already sent follow request
      */
-    private void addUserToFollowingForUserInDatabase(String userId, String followedUserId) {
-        DocumentReference userRef = _db.collection(USERS_COLLECTION_PATH).document(userId);
-        userRef.update(FOLLOWING_FIELD, FieldValue.arrayUnion(followedUserId));
+    private boolean searchInFollowerList(String followedId, String followerId) {
+        DocumentReference userDocument = _db.collection("Users").document(followedId);
+        Task<DocumentSnapshot> querySnapshotTask = userDocument.get();
+
+        while (!querySnapshotTask.isComplete());
+        Map<String, Object> objectMap = querySnapshotTask.getResult().getData();
+        ArrayList<String> requestList = (ArrayList<String>) objectMap.get(FOLLOWERS_FIELD);
+        if (requestList != null && !requestList.isEmpty()) {
+            return requestList.contains(followerId);
+        }
+        return false;
     }
 
     /**
-     * This method is responsible for adding a new user to the follower array of the given user
-     * @param userId the user performing the operation
-     * @param newFollowerId the user that is now a new follower of the given user
+     * Adds a user to the list of users requesting to follow another user
+     * @param followerUserId user requesting to follow
+     * @param followedUserId user following
      */
-    private void addNewFollowerForUserInDatabase(String userId, String newFollowerId) {
-        DocumentReference userRef = _db.collection(USERS_COLLECTION_PATH).document(userId);
-        userRef.update(FOLLOWERS_FIELD, FieldValue.arrayUnion(newFollowerId));
+    private void addUserToRequestsListInDatabase(String followerUserId, String followedUserId) {
+        DocumentReference userRef = _db.collection(USERS_COLLECTION_PATH).document(followedUserId);
+        userRef.update(FOLLOW_REQUEST_FIELD, FieldValue.arrayUnion(followerUserId));
```

Comment:
This request simply adds the userId to a new field FollowRequests of the target user, which is simply an array of all the users requsting to follow that user.